File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = []
    indexs = []
    for i in range(0, len(priorities)):
        indexs.append(i)
    while priorities:
        run = True
        temp = priorities.pop(0)
        temp_index = indexs.pop(0)
        for i in range(0, len(priorities)):
            if priorities[i] > temp:
                priorities.append(temp)
                indexs.append(temp_index)
                run = False
                break
        if run:
            answer.append(temp_index)
    return answer.index(location) + 1

File: 프로그래머스/2/42587. 프로세스/프로세스.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var count = 0
    var priorityQueue = priorities
    var targetIndex = location
    while priorityQueue.count != 0 {
        let max = priorityQueue.max() ?? 0
        let temp = priorityQueue.removeFirst()
        print(targetIndex)
        if temp >= max {
            count += 1
            if targetIndex == 0 {
                return count
            }
            targetIndex -= 1
        }
        else {
            priorityQueue.append(temp)
            targetIndex = targetIndex > 0 ? targetIndex - 1 : priorityQueue.count - 1
        }
        
    }
    return -1
}